src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -226,11 +233,62 @@ class PeerActor(
     context become waitingForConnectionResult(newConnection, noRetries)
   }
 
+  def handleEvmMptFastDownload(rlpxConnection: RLPxConnection): Receive = {
```

Comment:
I done it this way because we have one PeerActor per every peer so every peer can ask for fast download elements and requests will be handled in parallel. Also, all logic is encapsulated in a separate method.
About using futures for loading data: this functionality will be used only when a peer is doing fast download so for peer it is not important to get other information before he finishes fast download, but if we decide that it is required It should be easy to add.